hclsyntax/expression_ops.go

```diff
@@ -161,6 +195,8 @@ func (e *BinaryOpExpr) Value(ctx *hcl.EvalContext) (cty.Value, hcl.Diagnostics)
 			EvalContext: ctx,
 		})
 	}
+
+	givenRHSVal, rhsDiags := e.RHS.Value(ctx)
```

Comment:
Apologies if this is a basic question! 😅 I was wondering. Doesn't this already evaluate the RHS expression? I thought that if the goal is to short-circuit, we might want to avoid evaluating the RHS altogether if the LHS is sufficient to decide.

Reply:
Ah, I just read Martin's [comment](https://github.com/hashicorp/hcl/pull/713#issuecomment-2512188707) explaining the oddities that led to this. I see now that both sides need to be evaluated to retrieve the marks and unknown status. I think I got hung up on the term `short-circuiting` and how it typically works in other languages. It makes sense now why that's not an option here.